ingress/loadbalancer.go

```diff
@@ -17,19 +37,181 @@ type LoadBalancerEntry struct {
 	ServicePort int32
 }
 
-type noopLoadBalancer struct {
+// Nginx implementation
+type nginxLoadBalancer struct {
+	binary    string
+	confDir   string
+	port      int
+	cmd       *exec.Cmd
+	signaller Signaller
 }
 
-func (lb *noopLoadBalancer) Update(entries []LoadBalancerEntry) error {
-	log.Debugf("Updating loadbalancer for %v", entries)
-	return nil
+// Used for generating nginx cofnig
+type loadBalancerTemplate struct {
+	Port       int
+	Entries    []LoadBalancerEntry
+	WorkingDir string
```

Comment:
yeah the conf dir is used for the pid as well, I finally remember why I called it working dir :)